src/core/util/agent.cpp

```diff
@@ -62,10 +62,7 @@
 /* Force system call */
 #define sys_call(_result, _func, ...)                                                              \
     do {                                                                                           \
-        if (orig_os_api._func)                                                                     \
-            _result = orig_os_api._func(__VA_ARGS__);                                              \
-        else                                                                                       \
-            _result = ::_func(__VA_ARGS__);                                                        \
+        _result = SYSCALL(_func, __VA_ARGS__);                                                     \
```

Comment:
SYSCALL will not call the ::func version if orig funcs not found...

Reply:
That's an error.
I'll revert the code.